src/containers/NetworkViewers/RelationshipTreeViewer.tsx

```diff
@@ -197,37 +216,53 @@ class TreeViewer extends Component<Props, State> {
       if (data.nodes.length !== 0) {
         this.loadMissingResources(data.nodes);
         // eslint-disable-next-line react/no-did-update-set-state
-        setTimeout(() => this.setState({ loading: false }), 500);
+        setTimeout(() => {
+          this.setState({ loading: false });
+          // add collision force
+          this.forceGraphRef.current!.d3Force(
+            'collision',
+            // @ts-ignore
+            d3.forceCollide(node => Math.sqrt(100 / (node.level + 1)))
+          );
+          this.forceGraphRef.current!.d3Force(
+            'charge',
+            // @ts-ignore
+            d3.forceManyBody().strength(-80)
+          );
+        }, 500);
       }
     }
+    if (prevProps.asset === undefined && this.props.asset) {
+      this.props.fetchRelationshipsForAssetId(this.props.asset);
+    }
+    if (
+      prevProps.asset &&
+      this.props.asset &&
+      prevProps.asset.id !== this.props.asset.id
+    ) {
+      this.props.fetchRelationshipsForAssetId(this.props.asset);
+    }
     if (
       this.props.app.assetId &&
       prevProps.app.assetId !== this.props.app.assetId
     ) {
-      this.props.fetchRelationshipsForAssetId(this.props.app.assetId);
-      // add collision force
-      this.forceGraphRef.current!.d3Force(
-        'collision',
-        // @ts-ignore
-        d3.forceCollide(node => Math.sqrt(100 / (node.level + 1)))
-      );
-      this.forceGraphRef.current!.d3Force(
-        'charge',
-        // @ts-ignore
-        d3.forceManyBody().strength(-80)
-      );
+      this.props.fetchAssets([{ id: this.props.app.assetId }]);
     }
   }
 
   loadMissingResources = (nodes: any[]) => {
     const ids = nodes.reduce(
       (
-        prev: { timeseries: number[]; assets: number[]; models: number[] },
+        prev: { timeseries: IdEither[]; assets: IdEither[]; models: number[] },
         node
       ) => {
         switch (node.resource) {
           case 'asset': {
-            prev.assets.push(Number(node.resourceId));
+            if (isNumeric(node.resourceId)) {
```

Comment:
External ids can be numbers only of course (as a string). This is safe anyway?

Reply:
It is absolutely not safe :P its an assumption we've made, since internal id is going away soon anyways, i figure this is probably better.

Since the API does that dumb thing where if something fails, it returns nothing, debugging and refetching is not worth it

Reply:
Ok got it :)